diff --git a/drivers/media/i2c/ov5693.c b/drivers/media/i2c/ov5693.c
index 4cc796bbee92..6107c51cc7ff 100644
--- a/drivers/media/i2c/ov5693.c
+++ b/drivers/media/i2c/ov5693.c
@@ -88,7 +88,7 @@
 #define OV5693_FORMAT1_REG			CCI_REG8(0x3820)
 #define OV5693_FORMAT1_FLIP_VERT_ISP_EN		BIT(6)
 #define OV5693_FORMAT1_FLIP_VERT_SENSOR_EN	BIT(1)
-#define OV5693_FORMAT1_VBIN_EN			BIT(0)
+#define OV5693_FORMAT1_VBIN_EN			BIT(2)
 #define OV5693_FORMAT2_REG			CCI_REG8(0x3821)
 #define OV5693_FORMAT2_HDR_EN			BIT(7)
 #define OV5693_FORMAT2_FLIP_HORZ_ISP_EN		BIT(2)
@@ -316,7 +316,9 @@ static const struct cci_reg_sequence ov5693_global_regs[] = {
 	{CCI_REG8(0x5e10), 0x0c},
 	{CCI_REG8(0x3820), 0x00},
 	{CCI_REG8(0x3821), 0x1e},
-	{CCI_REG8(0x5041), 0x14}
+	{CCI_REG8(0x5041), 0x14},
+	/* Enable MIPI clock gate */
+	{CCI_REG8(0x4800), 0x20}
 };
 
 static const struct v4l2_rect ov5693_default_crop = {
@@ -376,7 +378,7 @@ static int ov5693_flip_horz_configure(struct ov5693_device *ov5693,
 	int ret;
 
 	ret = cci_update_bits(ov5693->regmap, OV5693_FORMAT2_REG, bits,
-			      enable ? bits : 0, NULL);
+			      enable ? 0 : bits, NULL);
 	if (ret)
 		return ret;
 
@@ -559,7 +561,7 @@ static int ov5693_mode_configure(struct ov5693_device *ov5693)
 		  &ret);
 
 	/* Offset X */
-	cci_write(ov5693->regmap, OV5693_OFFSET_START_X_REG, 0, &ret);
+	cci_write(ov5693->regmap, OV5693_OFFSET_START_X_REG, 2, &ret);
 
 	/* Output Size X */
 	cci_write(ov5693->regmap, OV5693_OUTPUT_SIZE_X_REG, mode->format.width,
@@ -567,7 +569,7 @@ static int ov5693_mode_configure(struct ov5693_device *ov5693)
 
 	/* Crop End X */
 	cci_write(ov5693->regmap, OV5693_CROP_END_X_REG,
-		  mode->crop.left + mode->crop.width, &ret);
+		  mode->crop.left + mode->crop.width + 7, &ret);
 
 	/* Horizontal Total Size */
 	cci_write(ov5693->regmap, OV5693_TIMING_HTS_REG, OV5693_FIXED_PPL,
@@ -578,7 +580,7 @@ static int ov5693_mode_configure(struct ov5693_device *ov5693)
 		  &ret);
 
 	/* Offset Y */
-	cci_write(ov5693->regmap, OV5693_OFFSET_START_Y_REG, 0, &ret);
+	cci_write(ov5693->regmap, OV5693_OFFSET_START_Y_REG, 2, &ret);
 
 	/* Output Size Y */
 	cci_write(ov5693->regmap, OV5693_OUTPUT_SIZE_Y_REG, mode->format.height,
@@ -586,7 +588,7 @@ static int ov5693_mode_configure(struct ov5693_device *ov5693)
 
 	/* Crop End Y */
 	cci_write(ov5693->regmap, OV5693_CROP_END_Y_REG,
-		  mode->crop.top + mode->crop.height, &ret);
+		  mode->crop.top + mode->crop.height + 7, &ret);
 
 	/* Subsample X increase */
 	cci_write(ov5693->regmap, OV5693_SUB_INC_X_REG,
@@ -813,7 +815,7 @@ static int ov5693_set_fmt(struct v4l2_subdev *sd,
 	struct ov5693_device *ov5693 = to_ov5693_sensor(sd);
 	const struct v4l2_rect *crop;
 	struct v4l2_mbus_framefmt *fmt;
-	unsigned int hratio, vratio;
+	unsigned int hratio, vratio, ratio;
 	unsigned int width, height;
 	unsigned int hblank;
 	int exposure_max;
@@ -831,18 +833,19 @@ static int ov5693_set_fmt(struct v4l2_subdev *sd,
 
 	/*
 	 * We can only support setting either the dimensions of the crop rect
-	 * or those dimensions binned (separately) by a factor of two.
+	 * or those dimensions binned by a factor of two.
 	 */
 	hratio = clamp_t(unsigned int,
 			 DIV_ROUND_CLOSEST(crop->width, width), 1, 2);
 	vratio = clamp_t(unsigned int,
 			 DIV_ROUND_CLOSEST(crop->height, height), 1, 2);
+	ratio = min_t(unsigned int, hratio, vratio);
 
 	fmt = __ov5693_get_pad_format(ov5693, state, format->pad,
 				      format->which);
 
-	fmt->width = crop->width / hratio;
-	fmt->height = crop->height / vratio;
+	fmt->width = crop->width / ratio;
+	fmt->height = crop->height / ratio;
 	fmt->code = MEDIA_BUS_FMT_SBGGR10_1X10;
 
 	format->format = *fmt;
@@ -852,10 +855,10 @@ static int ov5693_set_fmt(struct v4l2_subdev *sd,
 
 	mutex_lock(&ov5693->lock);
 
-	ov5693->mode.binning_x = hratio > 1;
-	ov5693->mode.inc_x_odd = hratio > 1 ? 3 : 1;
-	ov5693->mode.binning_y = vratio > 1;
-	ov5693->mode.inc_y_odd = vratio > 1 ? 3 : 1;
+	ov5693->mode.binning_x = ratio > 1;
+	ov5693->mode.inc_x_odd = ratio > 1 ? 3 : 1;
+	ov5693->mode.binning_y = ratio > 1;
+	ov5693->mode.inc_y_odd = ratio > 1 ? 3 : 1;
 
 	ov5693->mode.vts = __ov5693_calc_vts(fmt->height);
 
@@ -1396,6 +1399,7 @@ static const struct dev_pm_ops ov5693_pm_ops = {
 
 static const struct acpi_device_id ov5693_acpi_match[] = {
 	{"INT33BE"},
+	{"OVTI5693"},
 	{},
 };
 MODULE_DEVICE_TABLE(acpi, ov5693_acpi_match);
diff --git a/drivers/media/pci/intel/ipu-bridge.c b/drivers/media/pci/intel/ipu-bridge.c
index 510c0eca4484..27d90cd0e0cf 100644
--- a/drivers/media/pci/intel/ipu-bridge.c
+++ b/drivers/media/pci/intel/ipu-bridge.c
@@ -59,6 +59,7 @@ static const struct ipu_sensor_config ipu_supported_sensors[] = {
 	IPU_SENSOR_CONFIG("INT0310", 1, 55692000),
 	/* Omnivision OV5693 */
 	IPU_SENSOR_CONFIG("INT33BE", 1, 419200000),
+	IPU_SENSOR_CONFIG("OVTI5693", 1, 419200000),
 	/* Onsemi MT9M114 */
 	IPU_SENSOR_CONFIG("INT33F0", 1, 384000000),
 	/* Omnivision OV2740 */
@@ -134,6 +135,13 @@ static const struct dmi_system_id upside_down_sensor_dmi_ids[] = {
 		},
 		.driver_data = "OVTI02C1",
 	},
+	{
+		.matches = {
+			DMI_EXACT_MATCH(DMI_SYS_VENDOR, " Microsoft Corporation"),
+			DMI_EXACT_MATCH(DMI_PRODUCT_SKU, "Surface_Pro_9_2038"),
+		},
+		.driver_data = "OVTI5693",
+	},
 	{} /* Terminating entry */
 };
 
