diff --git a/drivers/media/i2c/ov13858.c b/drivers/media/i2c/ov13858.c
index 162b490469..45f99220af 100644
--- a/drivers/media/i2c/ov13858.c
+++ b/drivers/media/i2c/ov13858.c
@@ -3,6 +3,7 @@
 
 #include <linux/acpi.h>
 #include <linux/clk.h>
+#include <linux/delay.h>
 #include <linux/i2c.h>
 #include <linux/module.h>
 #include <linux/pm_runtime.h>
@@ -81,6 +82,9 @@
 #define OV13858_TEST_PATTERN_ENABLE	BIT(7)
 #define OV13858_TEST_PATTERN_MASK	0xfc
 
+/* Flip and Mirror Control */
+#define OV13858_REG_FORMAT1		0x3820
+
 /* Number of frames to skip */
 #define OV13858_NUM_OF_SKIP_FRAMES	2
 
@@ -937,6 +941,7 @@ static const char * const ov13858_test_pattern_menu[] = {
 };
 
 /* Configurations for supported link frequencies */
+#define OV13858_XVCLK_FREQ 19200000
 #define OV13858_NUM_OF_LINK_FREQS	2
 #define OV13858_LINK_FREQ_540MHZ	540000000ULL
 #define OV13858_LINK_FREQ_270MHZ	270000000ULL
@@ -1028,6 +1033,13 @@ static const struct ov13858_mode supported_modes[] = {
 	}
 };
 
+static const char * const ov13858_supply_names[] = {
+	"avdd",		/* Analog power */
+	"dvdd",		/* Digital circuit power */
+};
+
+#define OV13858_NUM_SUPPLIES	ARRAY_SIZE(ov13858_supply_names)
+
 struct ov13858 {
 	struct device *dev;
 	struct clk *clk;
@@ -1046,6 +1058,10 @@ struct ov13858 {
 	/* Current mode */
 	const struct ov13858_mode *cur_mode;
 
+	struct gpio_desc *reset;
+	struct regulator_bulk_data supplies[OV13858_NUM_SUPPLIES];
+	struct clk *xvclk;
+
 	/* Mutex for serialized access */
 	struct mutex mutex;
 };
@@ -1208,6 +1224,38 @@ static int ov13858_enable_test_pattern(struct ov13858 *ov13858, u32 pattern)
 				 OV13858_REG_VALUE_08BIT, val);
 }
 
+static int ov13858_set_ctrl_hflip(struct ov13858 *ov13858, u32 enable)
+{
+	int ret;
+	u32 val;
+
+	ret = ov13858_read_reg(ov13858, OV13858_REG_FORMAT1,
+			       OV13858_REG_VALUE_08BIT, &val);
+	if (ret)
+		return ret;
+
+	return ov13858_write_reg(ov13858, OV13858_REG_FORMAT1,
+				OV13858_REG_VALUE_08BIT,
+				enable ? (val & ~BIT(3)) : (val | BIT(3)));
+
+}
+
+static int ov13858_set_ctrl_vflip(struct ov13858 *ov13858, u32 enable)
+{
+	int ret;
+	u32 val;
+
+	ret = ov13858_read_reg(ov13858, OV13858_REG_FORMAT1,
+			       OV13858_REG_VALUE_08BIT, &val);
+	if (ret)
+		return ret;
+
+	return ov13858_write_reg(ov13858, OV13858_REG_FORMAT1,
+				OV13858_REG_VALUE_08BIT,
+				enable ? val | BIT(4) : val & ~BIT(4));
+
+}
+
 static int ov13858_set_ctrl(struct v4l2_ctrl *ctrl)
 {
 	struct ov13858 *ov13858 = container_of(ctrl->handler,
@@ -1257,10 +1305,14 @@ static int ov13858_set_ctrl(struct v4l2_ctrl *ctrl)
 	case V4L2_CID_TEST_PATTERN:
 		ret = ov13858_enable_test_pattern(ov13858, ctrl->val);
 		break;
+	case V4L2_CID_HFLIP:
+		ov13858_set_ctrl_hflip(ov13858, ctrl->val);
+		break;
+	case V4L2_CID_VFLIP:
+		ov13858_set_ctrl_vflip(ov13858, ctrl->val);
+		break;
 	default:
-		dev_info(ov13858->dev,
-			 "ctrl(id:0x%x,val:0x%x) is not handled\n",
-			 ctrl->id, ctrl->val);
+		ret = -EINVAL;
 		break;
 	}
 
@@ -1400,6 +1452,30 @@ ov13858_set_pad_format(struct v4l2_subdev *sd,
 	return 0;
 }
 
+static int ov13858_get_selection(struct v4l2_subdev *sd,
+				struct v4l2_subdev_state *state,
+				struct v4l2_subdev_selection *sel)
+{
+	struct ov13858 *ov13858 = to_ov13858(sd);
+	const struct ov13858_mode *mode = ov13858->cur_mode;
+
+	switch (sel->target) {
+	case V4L2_SEL_TGT_CROP:
+	case V4L2_SEL_TGT_CROP_BOUNDS:
+	case V4L2_SEL_TGT_CROP_DEFAULT:
+	case V4L2_SEL_TGT_NATIVE_SIZE:
+		sel->r.top = 0;
+		sel->r.left = 0;
+		sel->r.width = mode->width;
+		sel->r.height = mode->height;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
 static int ov13858_get_skip_frames(struct v4l2_subdev *sd, u32 *frames)
 {
 	*frames = OV13858_NUM_OF_SKIP_FRAMES;
@@ -1526,6 +1602,7 @@ static const struct v4l2_subdev_pad_ops ov13858_pad_ops = {
 	.enum_mbus_code = ov13858_enum_mbus_code,
 	.get_fmt = ov13858_get_pad_format,
 	.set_fmt = ov13858_set_pad_format,
+	.get_selection = ov13858_get_selection,
 	.enum_frame_size = ov13858_enum_frame_size,
 };
 
@@ -1622,6 +1699,12 @@ static int ov13858_init_controls(struct ov13858 *ov13858)
 				     V4L2_CID_TEST_PATTERN,
 				     ARRAY_SIZE(ov13858_test_pattern_menu) - 1,
 				     0, 0, ov13858_test_pattern_menu);
+
+	v4l2_ctrl_new_std(ctrl_hdlr, &ov13858_ctrl_ops,
+			  V4L2_CID_HFLIP, 0, 1, 1, 0);
+	v4l2_ctrl_new_std(ctrl_hdlr, &ov13858_ctrl_ops,
+			  V4L2_CID_VFLIP, 0, 1, 1, 0);
+
 	if (ctrl_hdlr->error) {
 		ret = ctrl_hdlr->error;
 		dev_err(ov13858->dev, "%s control init failed (%d)\n",
@@ -1655,6 +1738,90 @@ static void ov13858_free_controls(struct ov13858 *ov13858)
 	mutex_destroy(&ov13858->mutex);
 }
 
+static int ov13858_configure_gpios(struct ov13858 *ov13858)
+{
+	ov13858->reset = devm_gpiod_get_optional(ov13858->dev, "reset",
+						GPIOD_OUT_HIGH);
+	if (IS_ERR(ov13858->reset)) {
+		dev_err(ov13858->dev, "Error fetching reset GPIO\n");
+		return PTR_ERR(ov13858->reset);
+	}
+
+	return 0;
+}
+
+static int ov13858_get_regulators(struct ov13858 *ov13858)
+{
+	unsigned int i;
+
+	for (i = 0; i < OV13858_NUM_SUPPLIES; i++)
+		ov13858->supplies[i].supply = ov13858_supply_names[i];
+
+	return devm_regulator_bulk_get(ov13858->dev, OV13858_NUM_SUPPLIES,
+				       ov13858->supplies);
+}
+
+static void ov13858_sensor_powerdown(struct ov13858 *ov13858)
+{
+	gpiod_set_value_cansleep(ov13858->reset, 1);
+
+	regulator_bulk_disable(OV13858_NUM_SUPPLIES, ov13858->supplies);
+
+	clk_disable_unprepare(ov13858->xvclk);
+}
+
+static int ov13858_sensor_powerup(struct ov13858 *ov13858)
+{
+	int ret;
+
+	gpiod_set_value_cansleep(ov13858->reset, 1);
+
+	ret = clk_prepare_enable(ov13858->xvclk);
+	if (ret) {
+		dev_err(ov13858->dev, "Failed to enable clk\n");
+		goto fail_power;
+	}
+
+	ret = regulator_bulk_enable(OV13858_NUM_SUPPLIES, ov13858->supplies);
+	if (ret) {
+		dev_err(ov13858->dev, "Failed to enable regulators\n");
+		goto fail_power;
+	}
+
+	gpiod_set_value_cansleep(ov13858->reset, 0);
+
+	usleep_range(5000, 7500);
+
+	return 0;
+
+fail_power:
+	ov13858_sensor_powerdown(ov13858);
+	return ret;
+}
+
+static int __maybe_unused ov13858_sensor_suspend(struct device *dev)
+{
+	struct v4l2_subdev *sd = dev_get_drvdata(dev);
+	struct ov13858 *ov13858 = to_ov13858(sd);
+
+	ov13858_sensor_powerdown(ov13858);
+
+	return 0;
+}
+
+static int __maybe_unused ov13858_sensor_resume(struct device *dev)
+{
+	struct v4l2_subdev *sd = dev_get_drvdata(dev);
+	struct ov13858 *ov13858 = to_ov13858(sd);
+	int ret;
+
+	ret = ov13858_sensor_powerup(ov13858);
+	if (ret)
+		return ret;
+
+	return 0;
+}
+
 static int ov13858_probe(struct i2c_client *client)
 {
 	struct ov13858 *ov13858;
@@ -1673,7 +1840,7 @@ static int ov13858_probe(struct i2c_client *client)
 				     "failed to get clock\n");
 
 	freq = clk_get_rate(ov13858->clk);
-	if (freq != 19200000)
+	if (freq != OV13858_XVCLK_FREQ)
 		return dev_err_probe(ov13858->dev, -EINVAL,
 				     "external clock %lu is not supported\n",
 				     freq);
@@ -1681,19 +1848,28 @@ static int ov13858_probe(struct i2c_client *client)
 	/* Initialize subdev */
 	v4l2_i2c_subdev_init(&ov13858->sd, client, &ov13858_subdev_ops);
 
-	/* Check module identity */
-	ret = ov13858_identify_module(ov13858);
-	if (ret) {
-		dev_err(ov13858->dev, "failed to find sensor: %d\n", ret);
+	/* External clock init */
+	ov13858->xvclk = devm_clk_get_optional(ov13858->dev, "xvclk");
+	if (IS_ERR(ov13858->xvclk))
+		return dev_err_probe(ov13858->dev, PTR_ERR(ov13858->xvclk),
+				     "failed to get xvclk: %ld\n",
+				     PTR_ERR(ov13858->xvclk));
+
+	ret = ov13858_configure_gpios(ov13858);
+	if (ret)
 		return ret;
-	}
+
+	ret = ov13858_get_regulators(ov13858);
+	if (ret)
+		return dev_err_probe(ov13858->dev, ret,
+				     "Error fetching regulators\n");
 
 	/* Set default mode to max resolution */
 	ov13858->cur_mode = &supported_modes[0];
 
 	ret = ov13858_init_controls(ov13858);
 	if (ret)
-		return ret;
+		goto err_powerdown;
 
 	/* Initialize subdev */
 	ov13858->sd.internal_ops = &ov13858_internal_ops;
@@ -1710,20 +1886,38 @@ static int ov13858_probe(struct i2c_client *client)
 		goto error_handler_free;
 	}
 
-	ret = v4l2_async_register_subdev_sensor(&ov13858->sd);
-	if (ret < 0)
+	ret = ov13858_sensor_powerup(ov13858);
+	if (ret)
 		goto error_media_entity;
 
-	/*
-	 * Device is already turned on by i2c-core with ACPI domain PM.
-	 * Enable runtime PM and turn off the device.
-	 */
+	/* Check module identity */
+	ret = ov13858_identify_module(ov13858);
+	if (ret) {
+		dev_err(ov13858->dev, "failed to find sensor: %d\n", ret);
+		goto err_powerdown;
+	}
+
 	pm_runtime_set_active(ov13858->dev);
+	pm_runtime_get_noresume(ov13858->dev);
 	pm_runtime_enable(ov13858->dev);
-	pm_runtime_idle(ov13858->dev);
+
+	ret = v4l2_async_register_subdev_sensor(&ov13858->sd);
+	if (ret < 0)
+		goto err_pm_runtime;
+
+	pm_runtime_set_autosuspend_delay(ov13858->dev, 1000);
+	pm_runtime_use_autosuspend(ov13858->dev);
+	pm_runtime_put_autosuspend(ov13858->dev);
 
 	return 0;
 
+err_pm_runtime:
+	pm_runtime_disable(ov13858->dev);
+	pm_runtime_put_noidle(ov13858->dev);
+
+err_powerdown:
+	ov13858_sensor_powerdown(ov13858);
+
 error_media_entity:
 	media_entity_cleanup(&ov13858->sd.entity);
 
@@ -1746,6 +1940,10 @@ static void ov13858_remove(struct i2c_client *client)
 	pm_runtime_disable(ov13858->dev);
 }
 
+static const struct dev_pm_ops ov13858_pm_ops = {
+	SET_RUNTIME_PM_OPS(ov13858_sensor_suspend, ov13858_sensor_resume, NULL)
+};
+
 static const struct i2c_device_id ov13858_id_table[] = {
 	{ "ov13858" },
 	{}
@@ -1766,6 +1964,7 @@ static struct i2c_driver ov13858_i2c_driver = {
 	.driver = {
 		.name = "ov13858",
 		.acpi_match_table = ACPI_PTR(ov13858_acpi_ids),
+		.pm = &ov13858_pm_ops,
 	},
 	.probe = ov13858_probe,
 	.remove = ov13858_remove,
diff --git a/drivers/media/i2c/ov5693.c b/drivers/media/i2c/ov5693.c
index 4cc796bbee..6107c51cc7 100644
--- a/drivers/media/i2c/ov5693.c
+++ b/drivers/media/i2c/ov5693.c
@@ -88,7 +88,7 @@
 #define OV5693_FORMAT1_REG			CCI_REG8(0x3820)
 #define OV5693_FORMAT1_FLIP_VERT_ISP_EN		BIT(6)
 #define OV5693_FORMAT1_FLIP_VERT_SENSOR_EN	BIT(1)
-#define OV5693_FORMAT1_VBIN_EN			BIT(0)
+#define OV5693_FORMAT1_VBIN_EN			BIT(2)
 #define OV5693_FORMAT2_REG			CCI_REG8(0x3821)
 #define OV5693_FORMAT2_HDR_EN			BIT(7)
 #define OV5693_FORMAT2_FLIP_HORZ_ISP_EN		BIT(2)
@@ -316,7 +316,9 @@ static const struct cci_reg_sequence ov5693_global_regs[] = {
 	{CCI_REG8(0x5e10), 0x0c},
 	{CCI_REG8(0x3820), 0x00},
 	{CCI_REG8(0x3821), 0x1e},
-	{CCI_REG8(0x5041), 0x14}
+	{CCI_REG8(0x5041), 0x14},
+	/* Enable MIPI clock gate */
+	{CCI_REG8(0x4800), 0x20}
 };
 
 static const struct v4l2_rect ov5693_default_crop = {
@@ -376,7 +378,7 @@ static int ov5693_flip_horz_configure(struct ov5693_device *ov5693,
 	int ret;
 
 	ret = cci_update_bits(ov5693->regmap, OV5693_FORMAT2_REG, bits,
-			      enable ? bits : 0, NULL);
+			      enable ? 0 : bits, NULL);
 	if (ret)
 		return ret;
 
@@ -559,7 +561,7 @@ static int ov5693_mode_configure(struct ov5693_device *ov5693)
 		  &ret);
 
 	/* Offset X */
-	cci_write(ov5693->regmap, OV5693_OFFSET_START_X_REG, 0, &ret);
+	cci_write(ov5693->regmap, OV5693_OFFSET_START_X_REG, 2, &ret);
 
 	/* Output Size X */
 	cci_write(ov5693->regmap, OV5693_OUTPUT_SIZE_X_REG, mode->format.width,
@@ -567,7 +569,7 @@ static int ov5693_mode_configure(struct ov5693_device *ov5693)
 
 	/* Crop End X */
 	cci_write(ov5693->regmap, OV5693_CROP_END_X_REG,
-		  mode->crop.left + mode->crop.width, &ret);
+		  mode->crop.left + mode->crop.width + 7, &ret);
 
 	/* Horizontal Total Size */
 	cci_write(ov5693->regmap, OV5693_TIMING_HTS_REG, OV5693_FIXED_PPL,
@@ -578,7 +580,7 @@ static int ov5693_mode_configure(struct ov5693_device *ov5693)
 		  &ret);
 
 	/* Offset Y */
-	cci_write(ov5693->regmap, OV5693_OFFSET_START_Y_REG, 0, &ret);
+	cci_write(ov5693->regmap, OV5693_OFFSET_START_Y_REG, 2, &ret);
 
 	/* Output Size Y */
 	cci_write(ov5693->regmap, OV5693_OUTPUT_SIZE_Y_REG, mode->format.height,
@@ -586,7 +588,7 @@ static int ov5693_mode_configure(struct ov5693_device *ov5693)
 
 	/* Crop End Y */
 	cci_write(ov5693->regmap, OV5693_CROP_END_Y_REG,
-		  mode->crop.top + mode->crop.height, &ret);
+		  mode->crop.top + mode->crop.height + 7, &ret);
 
 	/* Subsample X increase */
 	cci_write(ov5693->regmap, OV5693_SUB_INC_X_REG,
@@ -813,7 +815,7 @@ static int ov5693_set_fmt(struct v4l2_subdev *sd,
 	struct ov5693_device *ov5693 = to_ov5693_sensor(sd);
 	const struct v4l2_rect *crop;
 	struct v4l2_mbus_framefmt *fmt;
-	unsigned int hratio, vratio;
+	unsigned int hratio, vratio, ratio;
 	unsigned int width, height;
 	unsigned int hblank;
 	int exposure_max;
@@ -831,18 +833,19 @@ static int ov5693_set_fmt(struct v4l2_subdev *sd,
 
 	/*
 	 * We can only support setting either the dimensions of the crop rect
-	 * or those dimensions binned (separately) by a factor of two.
+	 * or those dimensions binned by a factor of two.
 	 */
 	hratio = clamp_t(unsigned int,
 			 DIV_ROUND_CLOSEST(crop->width, width), 1, 2);
 	vratio = clamp_t(unsigned int,
 			 DIV_ROUND_CLOSEST(crop->height, height), 1, 2);
+	ratio = min_t(unsigned int, hratio, vratio);
 
 	fmt = __ov5693_get_pad_format(ov5693, state, format->pad,
 				      format->which);
 
-	fmt->width = crop->width / hratio;
-	fmt->height = crop->height / vratio;
+	fmt->width = crop->width / ratio;
+	fmt->height = crop->height / ratio;
 	fmt->code = MEDIA_BUS_FMT_SBGGR10_1X10;
 
 	format->format = *fmt;
@@ -852,10 +855,10 @@ static int ov5693_set_fmt(struct v4l2_subdev *sd,
 
 	mutex_lock(&ov5693->lock);
 
-	ov5693->mode.binning_x = hratio > 1;
-	ov5693->mode.inc_x_odd = hratio > 1 ? 3 : 1;
-	ov5693->mode.binning_y = vratio > 1;
-	ov5693->mode.inc_y_odd = vratio > 1 ? 3 : 1;
+	ov5693->mode.binning_x = ratio > 1;
+	ov5693->mode.inc_x_odd = ratio > 1 ? 3 : 1;
+	ov5693->mode.binning_y = ratio > 1;
+	ov5693->mode.inc_y_odd = ratio > 1 ? 3 : 1;
 
 	ov5693->mode.vts = __ov5693_calc_vts(fmt->height);
 
@@ -1396,6 +1399,7 @@ static const struct dev_pm_ops ov5693_pm_ops = {
 
 static const struct acpi_device_id ov5693_acpi_match[] = {
 	{"INT33BE"},
+	{"OVTI5693"},
 	{},
 };
 MODULE_DEVICE_TABLE(acpi, ov5693_acpi_match);
diff --git a/drivers/media/pci/intel/ipu-bridge.c b/drivers/media/pci/intel/ipu-bridge.c
index b2b7100949..984aca6596 100644
--- a/drivers/media/pci/intel/ipu-bridge.c
+++ b/drivers/media/pci/intel/ipu-bridge.c
@@ -59,6 +59,7 @@ static const struct ipu_sensor_config ipu_supported_sensors[] = {
 	IPU_SENSOR_CONFIG("INT0310", 1, 55692000),
 	/* Omnivision OV5693 */
 	IPU_SENSOR_CONFIG("INT33BE", 1, 419200000),
+	IPU_SENSOR_CONFIG("OVTI5693", 1, 419200000),
 	/* Onsemi MT9M114 */
 	IPU_SENSOR_CONFIG("INT33F0", 1, 384000000),
 	/* Omnivision OV2740 */
@@ -86,6 +87,8 @@ static const struct ipu_sensor_config ipu_supported_sensors[] = {
 	IPU_SENSOR_CONFIG("OVTI08A1", 1, 500000000),
 	/* Omnivision OV08x40 */
 	IPU_SENSOR_CONFIG("OVTI08F4", 3, 400000000, 749000000, 800000000),
+	/* OmniVision OV13858 */
+	IPU_SENSOR_CONFIG("OVTID858", 2, 270000000, 540000000),
 	/* Omnivision OV13B10 */
 	IPU_SENSOR_CONFIG("OVTI13B1", 1, 560000000),
 	IPU_SENSOR_CONFIG("OVTIDB10", 1, 560000000),
@@ -99,6 +102,19 @@ static const struct ipu_sensor_config ipu_supported_sensors[] = {
 	IPU_SENSOR_CONFIG("XMCC0003", 1, 321468000),
 };
 
+struct sensor_rotation_quirk {
+	char* acpi_id;
+};
+
+static const struct sensor_rotation_quirk dell_xps_quirks[] = {
+	{ "OVTI02C1" },
+};
+
+static const struct sensor_rotation_quirk microsoft_surface_quirks[] = {
+	{ "OVTID858" },
+	{ "OVTI5693" },
+};
+
 /*
  * DMI matches for laptops which have their sensor mounted upside-down
  * without reporting a rotation of 180° in neither the SSDB nor the _PLD.
@@ -109,14 +125,21 @@ static const struct dmi_system_id upside_down_sensor_dmi_ids[] = {
 			DMI_EXACT_MATCH(DMI_SYS_VENDOR, "Dell Inc."),
 			DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "XPS 13 9350"),
 		},
-		.driver_data = "OVTI02C1",
+		.driver_data = (void *)dell_xps_quirks,
 	},
 	{
 		.matches = {
 			DMI_EXACT_MATCH(DMI_SYS_VENDOR, "Dell Inc."),
 			DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "XPS 16 9640"),
 		},
-		.driver_data = "OVTI02C1",
+		.driver_data = (void *)dell_xps_quirks,
+	},
+	{
+		.matches = {
+			DMI_EXACT_MATCH(DMI_SYS_VENDOR, " Microsoft Corporation"),
+			DMI_EXACT_MATCH(DMI_PRODUCT_SKU, "Surface_Pro_9_2038"),
+		},
+		.driver_data = (void *)microsoft_surface_quirks,
 	},
 	{} /* Terminating entry */
 };
@@ -274,8 +297,14 @@ static u32 ipu_bridge_parse_rotation(struct acpi_device *adev,
 	const struct dmi_system_id *dmi_id;
 
 	dmi_id = dmi_first_match(upside_down_sensor_dmi_ids);
-	if (dmi_id && acpi_dev_hid_match(adev, dmi_id->driver_data))
-		return 180;
+	if (dmi_id) {
+		struct sensor_rotation_quirk *quirk;
+
+		for (quirk = dmi_id->driver_data; quirk->acpi_id; quirk++) {
+			if (acpi_dev_hid_match(adev, quirk->acpi_id))
+				return 180;
+		}
+	}
 
 	switch (ssdb->degree) {
 	case IPU_SENSOR_ROTATION_NORMAL:
diff --git a/drivers/platform/x86/intel/int3472/discrete.c b/drivers/platform/x86/intel/int3472/discrete.c
index 1505fc3ef7..514c1e8612 100644
--- a/drivers/platform/x86/intel/int3472/discrete.c
+++ b/drivers/platform/x86/intel/int3472/discrete.c
@@ -224,6 +224,8 @@ static void int3472_get_con_id_and_polarity(struct int3472_discrete_device *int3
 		*gpio_flags = GPIO_ACTIVE_HIGH;
 		break;
 	case INT3472_GPIO_TYPE_HANDSHAKE:
+	case 0x08:
+	case 0x10:
 		*con_id = "dvdd";
 		*gpio_flags = GPIO_ACTIVE_HIGH;
 		/* Setups using a handshake pin need 25 ms enable delay */
@@ -333,6 +335,8 @@ static int skl_int3472_handle_gpio_resources(struct acpi_resource *ares,
 	case INT3472_GPIO_TYPE_PRIVACY_LED:
 	case INT3472_GPIO_TYPE_POWER_ENABLE:
 	case INT3472_GPIO_TYPE_HANDSHAKE:
+	case 0x08:
+	case 0x10:
 		gpio = skl_int3472_gpiod_get_from_temp_lookup(int3472, agpio, con_id, gpio_flags);
 		if (IS_ERR(gpio)) {
 			ret = PTR_ERR(gpio);
@@ -357,6 +361,8 @@ static int skl_int3472_handle_gpio_resources(struct acpi_resource *ares,
 			second_sensor = int3472->quirks.avdd_second_sensor;
 			fallthrough;
 		case INT3472_GPIO_TYPE_HANDSHAKE:
+		case 0x08:
+		case 0x10:
 			ret = skl_int3472_register_regulator(int3472, gpio, enable_time_us,
 							     con_id, second_sensor);
 			if (ret)
